File: backend/src/main/java/tko/model/dto/workout/WorkoutExerciseDTO.java
package tko.model.dto.workout;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class WorkoutExerciseDTO {
    private Long id;
    @NotNull
    private Long workoutId;
    @NotNull
    private Long exerciseId;
    @NotNull
    private Integer sets;
    @NotNull
    private Integer reps;
    @NotNull
    private Double distance;
    @NotNull
    private Double duration;
    @NotNull
    private Integer restTime;
    private Integer exerciseOrder;
}

File: app/src/main/java/tk/ssau/fitnesstko/model/dto/WorkoutExerciseDto.kt
package tk.ssau.fitnesstko.model.dto

data class WorkoutExerciseDto(
    val id: Long?,
    val workoutId: Long,
    val exerciseId: Long,
    val sets: Int,
    val reps: Int,
    val distance: Double,
    val duration: Double,
    val restTime: Int,
    val exerciseOrder: Int?
)